OverlayPlugin.Core/MemoryProcessors/Combatant/CombatantMemory61.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace RainbowMage.OverlayPlugin.MemoryProcessors.Combatant
+{
+    class CombatantMemory61 : CombatantMemory
+    {
+        private const string charmapSignature = "48c1ea0381faa9010000????8bc2488d0d";
+
+        public CombatantMemory61(TinyIoCContainer container)
+            : base(container, charmapSignature, CombatantMemory.Size, EffectMemory.Size)
+        {
+            
+        }
+        // Returns a combatant if the combatant is a mob or a PC.
+        protected override unsafe Combatant GetMobFromByteArray(byte[] source, uint mycharID)
+        {
+            fixed (byte* p = source)
+            {
+                CombatantMemory mem = *(CombatantMemory*)&p[0];
+                ObjectType type = (ObjectType)mem.Type;
+                if (mem.ID == 0 || mem.ID == emptyID)
+                    return null;
+            }
+            return GetCombatantFromByteArray(source, mycharID, false);
+        }
+
+        // Will return any kind of combatant, even if not a mob.
+        // This function always returns a combatant object, even if empty.
+        protected override unsafe Combatant GetCombatantFromByteArray(byte[] source, uint mycharID, bool isPlayer, bool exceptEffects = false)

```

Comment:
I wasn't really able to figure out a way to handle this with a generic function, because C# interfaces require only methods/properties, not fields, and I think that duplicating a getter/setter for every struct would end up being more duplicate code than this setup? I'm fine with other thoughts/suggestions here though.

Reply:
Yeah, I'm not sure there's a great answer here, but I thought I'd ask.

My only real thought is to do some sort of `Combatant GetCombatantFromByteArray<CombatantMemory>(etc) where CombatantMemory : unmanaged` and then use `dynamic mem = *(CombatantMemory*)&p[0];` to avoid having to re-write out all the properties, but that feels like a risky hack rather than using the type system more properly.  My kingdom for some C++ macros.  <_<

Reply:
I think I'd rather look into something like that as a followup PR regardless, since it seems more likely to cause errors than what's currently in place.